pkg/server/handler/v1alpha1/integration.go

```diff
@@ -55,7 +57,35 @@ func ListIntegrations(ctx context.Context, tenant string, pagination *types.Pagi
 		integrations = append(integrations, *integration)
 	}
 
-	return types.NewListResponse(int(size), integrations[pagination.Start:end]), nil
+	results := []api.Integration{}
+	// Handle filter.
+	if qp.Filter == "" {
+		results = integrations[qp.Start:end]
+	} else {
+		filterParts := strings.Split(qp.Filter, "=")
+		if len(filterParts) != 2 {
+			return nil, cerr.ErrorUnknownInternal.Error("Filter query param %s is not correct", qp.Filter)
+		}
+		fieldsStr, keyWord := filterParts[0], filterParts[1]
+		fields := strings.Split(fieldsStr, ".")
+
+		for i := qp.Start; i < end; i++ {
```

Comment:
We should apply filters to all items, and then perform pagination for the remained items.

Reply:
Done